src/workerd/util/sqlite-test.c++

```diff
@@ -509,7 +509,9 @@ KJ_TEST("SQLite read row counters (basic)") {
   constexpr int dbRowCount = 1000;
   auto insertStmt = db.prepare("INSERT INTO things (id, unindexed_int, value) VALUES (?, ?, ?)");
   for (int i = 0; i < dbRowCount; i++) {
-    insertStmt.run(i, i * 1000, kj::str("value", i));
+    auto query = insertStmt.run(i, i * 1000, kj::str("value", i));
```

Comment:
As far as I can tell, this new test code only tests the `Query(SqliteDatabase& db, const Regulator& regulator, Statement& statement, Params&&... bindings)` (on line 404 in sqlite.h).

You've also changed ` Query(SqliteDatabase& db, const Regulator& regulator, kj::StringPtr sqlCode, Params&&... bindings)` (from line 412).  This version of the constructor _is_ called by `countRowsTouched`.  How come the test was passing before adding the code?  I think it's because `countRowsTouched` makes a new implicit statement on every call, rather than reusing the statement.

Is there a test we can add that tickles the problem and is fixed by adding the `resetRowCounters();` statement?

Reply:
Looking at this a bit more... why do we have the `resetRowCounters()` in `Query(SqliteDatabase& db, const Regulator& regulator, kj::StringPtr sqlCode, Params&&... bindings)` (from line 412)?  We're creating a new statement from the SQL code.

Reply:
The `kj::StringPtr sqlCode` version always prepares a new statement from scratch hence its counts always start from zero anyway, so there was no bug in that version of the constructor.

Reply:
We don't need the reset counters in the constructor which creates the statement, only for the one that takes in a statement. I updated the code accordingly